src/DataFrame/Lazy/IO/CSV.hs

```diff
@@ -90,7 +90,7 @@ readSeparated c opts path = do
         Nothing ->
             countRows c path >>= \total -> if hasHeader opts then return (total - 1) else return total
         Just n -> if hasHeader opts then return (n - 1) else return n
-    let (begin, len) = case rowRange opts of
```

Comment:
This part of the code is almost entirely experimental and needs to be revisited. So fine to make edits that might break.